packages/react-app/components/TokenContract.tsx

```diff
@@ -0,0 +1,489 @@
+  );
+}
+
+const iconStyle = {
+  // backgroundColor: "#000000", // camel cased
+  // height: "100px",
+  // color: "yellow",
+  marginLeft: "0px",
+  marginRight: "0px",
+  marginTop: "10px",
+  marginBottom: "10px",
+};
+
+const buttonStyle = {
+  // backgroundColor: "#000000", // camel cased
+  // height: "100px",
+  // color: "yellow",
+  marginLeft: "0px",
+  marginRight: "5px",
+  marginTop: "20px",
+  marginBottom: "20px",
+};
+
+export function TokenContract({ contractData }) {
+  const { kit, address, network, performActions } = useContractKit();
+  const [tokenValue, setTokenValue] = useState();
+  const [tokenInput, setTokenInput] = useInput({ type: "text" });
+  const [contractLink, setContractLink] = useState("");
+  const { enqueueSnackbar, closeSnackbar } = useSnackbar();
+
+  //   const [{ data, error, loading }, getSigner] = useSigner();
+  const [tokenBalance, setTokenBalance] = useState("");
+  const [tokenName, setTokenName] = useState("");
+  const [tokenDecimals, setTokenDecimals] = useState("");
+  const [tokenInitialSupply, setTokenInitialSupply] = useState("");
+  const [tokenSymbol, setTokenSymbol] = useState("");
+  const [tokenTotal, setTokenTotal] = useState("");
+  const [tokenBalanceOf, setTokenBalanceOf] = useState("");
+  const [newAddress, setNewAddress] = useState("");
+  const [tokenTransferValue, setTokenTransferValue] = useState("");
+
+  const contract = contractData
+    ? new kit.web3.eth.Contract(contractData.abi, contractData.address)
+    : null;
+
+  useEffect(() => {
+    if (contractData) {
+      setContractLink(`${network.explorer}/address/${contractData.address}`);
+    }
+  }, [network, contractData]);
+
+  const setToken = async () => {
+    try {
+      await performActions(async (kit) => {
+        const gasLimit = await contract.methods.store(tokenInput).estimateGas();
+
+        const result = await contract.methods
+          .store(tokenInput)
+          .send({ from: address, gasLimit });
+
+        console.log(result);
+
+        const variant = result.status == true ? "success" : "error";
+        const url = `${network.explorer}/tx/${result.transactionHash}`;
+        const action = (key) => (
+          <>
+            <Link href={url} target="_blank">
+              View in Explorer
+            </Link>
+            <Button
+              onClick={() => {
+                closeSnackbar(key);
+              }}
+            >
+              X
+            </Button>
+          </>
+        );
+        enqueueSnackbar("Transaction sent", {
+          variant,
+          action,
+        });
+      });
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  const getTokenName = async () => {
+    const tokenName = await contract.methods.name().call();
+    setTokenName(tokenName);
+  };
+
+  const getTokenSymbol = async () => {
+    const tokenSymbol = await contract.methods.symbol().call();
+    setTokenSymbol(tokenSymbol);
+  };
+
+  const getTokenInitialSupply = async () => {
```

Comment:
The format in this one is a little weird, you are using await syntax and a Promise callback. I prefer the syntax you use in the `getTokenTotal` function. Both work, I think you should be consistent though.

Reply:
I like the `getTokenTotal` format but it wasn't working for these. It gives a big number error on the first line before giving the chance to convert in later lines. Would like to switch it back but could use help because I couldn't fix it.

Reply:
So weird I'll check what I was doing wrong. Updated both functions to use this and they're working now.